### ISSUE TITLE
Text not sprawling outside of image area

### ISSUE THREAD

**Issue by GarkGarcia** [Mathics3/mathics-django#2]
This is a follow up to https://github.com/mathics/Mathics/issues/527:

> Inserting spaces into `Text` causes nothing to be displayed:
> ![2016-09-02-155232_1920x1080_scrot](https://cloud.githubusercontent.com/assets/1217643/18193916/7851cc7c-7125-11e6-84d3-ed4be2967655.png)

**Comment by rocky** [comment]
I've tracked this down into a Mathics problem in Boxing/Formatting. 

```
  ('<math display="block"><mstyle mathvariant="sans-serif"><mglyph width="400px" '
 'height="119px" '
 'src="data:image/svg+xml;base64,CiAgICAgICAgICAgIDxzdmcgeG1sbnM6c3ZnPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgICAgICAgICAgICAgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgICAgICAgICAgICAgIHZlcnNpb249IjEuMSIKICAgICAgICAgICAgICAgIHZpZXdCb3g9IjE1Ny4wMDAwMDAgNDYuMDIzODEwIDg2LjAwMDAwMCAyNy4wMDAwMDAiPgogICAgICAgICAgICAgICAgPHRleHQgeD0iMjAwLjAiIHk9IjU5LjUyMzgwOTUyMzgwOTUyNiIgb3g9IjAiIG95PSIwIiBzdHlsZT0idGV4dC1hbmNob3I6bWlkZGxlOyBkb21pbmFudC1iYXNlbGluZTptaWRkbGU7IHN0cm9rZTogcmdiKDAuMDAwMDAwJSwgMC4wMDAwMDAlLCAwLjAwMDAwMCUpOyBzdHJva2Utb3BhY2l0eTogMTsgZmlsbDogcmdiKDAuMDAwMDAwJSwgMC4wMDAwMDAlLCAwLjAwMDAwMCUpOyBmaWxsLW9wYWNpdHk6IDE7IGNvbG9yOiByZ2IoMC4wMDAwMDAlLCAwLjAwMDAwMCUsIDAuMDAwMDAwJSk7IG9wYWNpdHk6IDEuMCI+SGVsbG93IFdvcmxkPC90ZXh0PgogICAgICAgICAgICA8L3N2Zz4KICAgICAgICA="/></mstyle></math>')
```

is getting converted to: 

```
<math xmlns="http://www.w3.org/1998/Math/MathML" display="block">
  <mstyle displaystyle="true">
    <mspace width="400pxpx" height="129pxpx" id="math_object_10" />
  </mstyle>
</math>
```

Notice the `pxpx` at the end of "height" and "width". 

@mmatera Know what it is that does the conversion. Is this on the Mathics side? The (Django) MathJax side? 

I wonder if we need to turn this into SVG at all.

**Comment by mmatera** [comment]
This is what I get 
![imagen](https://user-images.githubusercontent.com/2862288/120194488-0b422f80-c1f4-11eb-9d82-cb6109620144.png)

The problem is that the bounding box is not properly evaluated. However, I am not sure if the problem is in the mathics core or in the post-processing.

The idea of generating directly svg was implemented in PR #635 by using the node.js Webkit. The problem with that is that adds more dependencies. I imagine then two other ways to solve this problem
1) move all the code that generates the web output to a different Python package, that takes (let's say) a JSON representation of a ``Box`` ``Expression``, and generates an ``HTML`` output. Then use this in the client, and maybe also in the core/ a pymathics module for Exporting to HTML, SVG and other formats.
2) (my favorite) Implement the interface via ExportString. Then, you can set format rules directly in WL.

**Comment by rocky** [comment]
> The problem is that the bounding box is not properly evaluated. However, I am not sure if the problem is in the mathics core or in the post-processing.

I suspect it isn't core, but in post-processing. And if there is dumb post-processing of the kind I've seen before, this is probably wrong, and better would be to keep the structure in place when passing from Mathics core to the front ends.

> The idea of generating directly svg was implemented in PR #635 by using the node.js Webkit. The problem with that is that adds more dependencies. I imagine then two other ways to solve this problem

Ugh. What I am coming to learn is that in poke1024's code in moving things forward also moved things in a bad direction as well. And then he's not around to help out after we've swallowed the code, (and it is largely not commented either). 

> 1. move all the code that generates the web output to a different Python package, that takes (let's say) a JSON representation of a `Box Expression`, and generates an HTML output. Then use this in the client, and maybe also in the core/ a pymathics module for Exporting to HTML, SVG and other formats.

We discussed this a little, I think. Ideal is just to send back to the front-end a the Mathics Expression and let it make calls to do the Boxing and and Formatting. For mathics-django this can be as it is currently in Python. For Jupyter, we'd convert the Python to JSON which is trivial to do.

> 2. (my favorite) Implement the interface via `ExportString`. Then, you can set format rules directly in WL.

I am in favor of setting format rules in WL/Mathics. I don't see why `ExportString` is the only method for getting this done though. However as I have come to learn with so many other things in this code, it is such a heavy lift to fix things properly that I definitely can't do so in one go. Sigh. So this might be a fallback interim solution.

From what I have seen, `ExportString` is a little bit hacky because we don't have the right underlying primitives and concepts in place. 

And it is not a _string_ per say we want but the JSON for wire protocols

**Comment by rocky** [comment]
> This is what I get
> ![imagen](https://user-images.githubusercontent.com/2862288/120194488-0b422f80-c1f4-11eb-9d82-cb6109620144.png)
> 
> The problem is that the bounding box is not properly evaluated. However, I am not sure if the problem is in the mathics core or in the post-processing.
> 

I looked at this in more detail. A fix is to specify the text size. Somehow whatever the pointsize used in the calculation of the bounding box doesn't match the default pointsize that svg will select. If you set the point size to 10px, then things work fine.

More serious is that this the SVG inset_box routine didn't even have a way to set the font size or font parameters. And this is part of a bigger problem with most of those routines. 

At least for me splitting this out into its own module has allowed me to focus on just the SVG aspect and I see how shoddy it kind of is. And this to me shows a bigger problem with the codebase. By trying to roll our own SVG libraries we do a pretty shoddy job.
